src/checkers/urlchecker.py

```diff
@@ -76,13 +76,21 @@ async def check(self, external_data):
         else:
             url = external_data.current_version.url
 
+        strip_query = external_data.checker_data.get("strip-query", False)
```

Comment:
I have a theory that, if we scanned all Flathub repos for URLs, we would find 0 examples of query parameters that matter. :)

But I guess false is the conservative default. Should be documented in README though.

Reply:
Added to readme, please check if the wording is ok.
>if we scanned all Flathub repos for URLs, we would find 0 examples of query parameters that matter

Well, that's actually doable. If your theory proves right, we can just invert the default value.

Reply:
I happened to have clones of all Flathub repos from about 6 weeks ago because I used them to prepare that talk.

I used the following program to scrape all the URLs:

```python3
#!/usr/bin/env python3
#
# dump-urls: Extract external URLs from the given manifests
#
# Copyright © 2021 Endless Mobile, Inc.
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License along
# with this program; if not, write to the Free Software Foundation, Inc.,
# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.

import argparse
import sys
from pathlib import Path

exe = Path(sys.argv[0]).resolve()
sys.path.insert(0, str(exe.parent))

from src.checker import ManifestChecker

def find_manifests(manifest_dirs):
    for manifest_dir in manifest_dirs:
        for extension in (".json", ".yaml", ".yml"):
            manifest_path = (manifest_dir / (manifest_dir.name + extension))
            if manifest_path.is_file():
                yield manifest_path
                break

if __name__ == "__main__":
    parser = argparse.ArgumentParser(
        description="Extract remote URLs from a list of Flathub repos",
    )
    parser.add_argument(
        "manifest_dirs", nargs="*", type=Path,
        help="path to a Flathub repo clone",
    )
    args = parser.parse_args()

    for manifest_path in find_manifests(args.manifest_dirs):
        checker = ManifestChecker(str(manifest_path))
        for x in checker.get_external_data():
            if x.type != x.Type.GIT:
                url = x.current_version.url
                if '?' in url:
                    print(manifest_path.name, url)
```

Here is the output:

```
com.github.tenderowl.norka.json https://downloads.sourceforge.net/project/gtkspell/3.0.10/gtkspell3-3.0.10.tar.xz?r=https%3A%2F%2Fsourceforge.net%2Fprojects%2Fgtkspell%2Ffiles%2F3.0.10%2Fgtkspell3-3.0.10.tar.xz%2Fdownload&ts=1572946840
```

And I confirmed that the URL redirects to the same place with or without the query string. So indeed I think the default can be the other way around.

Reply:
Noted, thanks for checking this. Still I'm hesitant to change the default behavior. Maybe flip the default after resolving #169, so that less apps would be affected by potential misbehavior?

Reply:
#169 is resolved so shall we change the default to "yes, strip the query part" and land this? I really value the default behaviour being what most users want to minimise the amount of gunk they have to blindly paste into their manifest, and to date we have seen many cases where this behaviour is desirable and none where it is not.

Reply:
If make stripping query the default, maybe we should do it for all checker, not just the rotating-url one?